DOCS-3541 Service Catalog (beta) docs (#14152)
DataDog/documentation: content/en/tracing/faq/service_catalog.md

---
title: Datadog Service Catalog
kind: faq
is_beta: true
further_reading:
- link: "/tracing/faq/service_definition_api/"
  tag: "Documentation"
  text: "Service Definition API"
---

{{< img src="tracing/service_catalog/service_catalog.mp4" video=true alt="Navigating around the Service Catalog" style="width:100%;" >}}

<div class="alert alert-warning">This feature is in private beta and the endpoints are likely to change.</div>

## Overview

> Datadog Service Catalog is a centralized place to access important information about all services in your organization. Achieve end-to-end service ownership at scale, get real-time performance insights, detect and address reliability and security risks, and manage application dependencies all in one place. Pivot quickly into team communications tools (Slack), source control (Github), dashboards, and Datadog views that receive and monitor telemetry data for each service.

Author: Took some of these, but some changed the meaning incorrectly.

- Speeding incident recovery by increasing confidence and simplifying locating owners of upstream and downstream services and dependencies.
- Detecting which services aren't reporting observability data or having that data monitored.
- Facilitating the practice of good tagging to optimize cross-telemetry insights.
- Providing engineering leadership with a high-level view of reliability practices across teams and services. 
- Spotting issues like missing SLOs, monitors, or services without ownership.

## Browse the Service Catalog

On the [Service Catalog page][1], see the list of services in your Datadog organization, either detected from collected data, or defined by someone [registering the service](#registering-a-new-service). To find a particular service, search by its name. To filter the list, select one or more facets. For example, to see detected services that don't yet have a service definition, click the **Ownership Info > Telemetry Only** facet. You might find it helpful to filter by your team name. Or scope the metrics shown to particular environments and clusters to see only matching services in the list.

> The list of services in the catalog is sortable service type, service name, as well as many of the other columns, making it quick to find specific or missing details (for example owning team on the Ownership view), and to sort by urgency (for example, number of triggered monitors on the Reliability view).

Author: ```suggestion
The Service Catalog list is sortable by service type, service name, and many of the other columns. You can find missing ownership by sorting by team in the Ownership view and looking for blanks. Or you can sort by urgency in the Reliability view and see services with the most triggered monitors.
```

The Telemetry column instantly shows you what types of telemetry data Datadog is collecting for the service, and clicking the icons takes you into the corresponding Datadog product view for the service. For example, the Agent has sent traces to Datadog, click the Traces icon to view them in APM.

Clicking the three-dot menu lets you edit the service definition, if it exists, or link to one if the service hasn't yet been defined. You might have to first [set up an integration with your source code system][2] if you haven't yet.

Sort the table by the Team or On Call columns to quickly spot services where ownership and responsibility have not been specified, or to see which services each team is responsible for.

### Reliability view

The Reliability view brings together key information about the stability of your services. Sort the table by clicking on columns to reveal:

- Which services deployed most recently, or have not deployed for a long time.

> - Which services are reporting the most errors, and are they new issues.

Author: ```suggestion
- Which services are reporting the most errors, and whether they are new issues.
```

Service definitions include the following elements, all optional except the service name:

Service name
: An identifier for the service, unique within Datadog. By default, this is the `DD_SERVICE` value from the incoming data.

Team
: Your name for the team responsible for developing and maintaining the service.

Contacts
: One or more ways to contact the team, such as Slack channels or email addresses.

Links
: A list of links to important resources for the service, such as runbooks.

Repos
: A list of source control repositories that contain source code and related files for maintaining, testing, and deploying the service.

Docs
: Links to documentation for the service.

Tags
: Like all Datadog tags, these help you find, filter, and group whatever they are applied to, in this case service definitions.

Integrations
: Custom strings for hooking up to integrations such as PagerDuty for identifying on-call for the service.

### Enriching a listed service 

> Initially, services that are sending data to Datadog through one of the telemetry products (for example, traces, logs, profiles, infrastructure, or network device monitoring) are listed in the Service Catalog with an `UNDEFINED` label. The `UNDEFINED` label means that no service definition has been associated with that service yet. 

Reviewer: ```suggestion
Initially, services sending data to Datadog through one of the telemetry products (Traces, Logs, Profiles, Infrastructure, Network Performance, and RUM) are listed with an `UNDEFINED` label, which means that no service definition has been associated with the service yet. 
```

> Initially, services that are sending data to Datadog through one of the telemetry products (for example, traces, logs, profiles, infrastructure, or network device monitoring) are listed in the Service Catalog with an `UNDEFINED` label. The `UNDEFINED` label means that no service definition has been associated with that service yet. 

Reviewer: ```suggestion
Initially, services sending data to Datadog through one of the telemetry products (Traces, Logs, Profiles, Infrastructure, Network Performance, and RUM) are listed with an `UNDEFINED` label, which means that no service definition has been associated with the service yet. 
```

> Add service ownership information such as team name, Slack channels, and source code repositories, by using the [Service Definition API][5].

Reviewer: ```suggestion
To add service ownership information such as team name, Slack channels, and source code repositories, use the [Service Definition API][5].
```

> Add service ownership information such as team name, Slack channels, and source code repositories, by using the [Service Definition API][5].

Reviewer: ```suggestion
To add service ownership information such as team name, Slack channels, and source code repositories, use the [Service Definition API][5].
```

### Registering a new service

> You can manage service ownership information even for services that are not emitting any Datadog telemetry. Specify service ownership, on-call info, and custom tags with the [Service Definition API][5], and the information is reflected in the Service Catalog UI. 

Reviewer: ```suggestion
You can also manage service ownership information for services that do not emit any Datadog telemetry. To register a service, specify the service ownership, on-call information, and custom tags using the [Service Definition API][5]. 
```

> You can manage service ownership information even for services that are not emitting any Datadog telemetry. Specify service ownership, on-call info, and custom tags with the [Service Definition API][5], and the information is reflected in the Service Catalog UI. 

Reviewer: ```suggestion
You can also manage service ownership information for services that do not emit any Datadog telemetry. To register a service, specify the service ownership, on-call information, and custom tags using the [Service Definition API][5]. 
```

> The following YAML shows an example service definition:

Reviewer: ```suggestion
See the example service definition YAML:
```

> The following YAML shows an example service definition:

Reviewer: ```suggestion
See the example service definition YAML:
```

```yaml
---
schema-version: v2
dd-service: product-recommendation-lite
team: Shopist
contacts:
  - type: slack
    contact: https://exampleco.slack.com/archives/S319HSDB32
links: 
  - name: Demo Dashboard
    type: dashboard
    url: https://app.datadoghq.com/dashboard/abc-def-ghi
repos: 
  - name: Source 
    provider: github 
    url: https://github.com/DataDog/shopist/tree/prod/product-recommendation-lite
  - name: Deployment 
    provider: github 
    url: https://github.com/DataDog/shopist/blob/prod/k8s/dd-trace-demo/templates/product-recommendation-lite-deployment.yaml
docs: 
  - name: Datadog Doc
    provider: link
    url: https://docs.datadoghq.com/tracing/faq/service_catalog/
tags: []
```

> You can register multiple services with a single YAML file by separating each service definition with a line containing three hyphens: `---`.

Reviewer: ```suggestion
To register multiple services in one YAML file, separate each service definition with a line containing three hyphens: `---`.
```

> You can register multiple services with a single YAML file by separating each service definition with a line containing three hyphens: `---`.

Reviewer: ```suggestion
To register multiple services in one YAML file, separate each service definition with a line containing three hyphens: `---`.
```

---
title: Datadog Service Catalog
kind: faq
is_beta: true
further_reading:
- link: "/tracing/faq/service_definition_api/"
  tag: "Documentation"
  text: "Service Definition API"
---

<div class="alert alert-warning">This feature is in private beta and the endpoints are likely to change.</div>

{{< img src="tracing/service_catalog/service_catalog.mp4" video=true alt="Navigating around the Service Catalog" style="width:100%;" >}}

## Overview

> Datadog Service Catalog is a centralized place to access important information about all services in your organization. Achieve end-to-end service ownership at scale, get real-time performance insights, detect and address reliability and security risks, and manage application dependencies all in one place. Access team communications tools such as Slack, source control such as Github, dashboards, and Datadog views that receive and monitor telemetry data for each service.

Reviewer: ```suggestion
Datadog Service Catalog is a centralized place to access important information about all services in your organization. Achieve end-to-end service ownership at scale, get real-time performance insights, detect and address reliability and security risks, and manage application dependencies all in one place. Access team communications tools such as Slack, source control such as GitHub, Datadog dashboards, and Datadog views that receive and monitor telemetry data for each service.
```
